src/app/_components/Header/NotificationList.tsx

```diff
@@ -0,0 +1,158 @@
+import { useInfiniteQuery } from '@tanstack/react-query';
+import Link from 'next/link';
+import { useMemo } from 'react';
+import {
+  NOTIFICATIONS_REF_OPTIONS,
+  NOTIFICATIONS_TAKE,
+} from '@/constants/constants';
+import useIntersect from '@/hooks/useIntersect';
+import useReadNotification from '@/hooks/useReadNotification';
+import { getNotifications } from '@/lib/apis/notifications';
+import { formatRelativeOrShortDate } from '@/utils/dateTimeUtils';
+import {
+  checkFilterOption,
+  generateNotificationLink,
+} from '@/utils/notificationUtils';
+import { userType } from '@/types/auth';
+import {
+  IGetNotifications,
+  NotificationsFilterOption,
+} from '@/types/notifications';
+
+const NotificationList = ({
+  userType,
+  alarmCount,
+}: {
+  userType: userType;
+  alarmCount: string;
+}) => {
+  const fetchNotifications = ({
+    pageParam,
+  }: {
+    pageParam: IGetNotifications | undefined;
+  }) =>
+    getNotifications({
+      pageSize: NOTIFICATIONS_TAKE,
+      filterOption: 'UNREAD',
+      ...pageParam,
+    });
+
+  const {
+    data: notificationsList,
+    hasNextPage,
+    isLoading,
+    fetchNextPage,
+    isFetchingNextPage,
+  } = useInfiniteQuery({
+    queryKey: ['notifications', 'UNREAD'],
+    queryFn: fetchNotifications,
+    initialPageParam: undefined,
+    getNextPageParam: (lastPage, allpages) => {
+      const currentCount = allpages.length * NOTIFICATIONS_TAKE;
+
+      return allpages[0]?.totalItemCount &&
+        allpages[0].totalItemCount > currentCount
+        ? {
+            pageSize: NOTIFICATIONS_TAKE,
+            filterOption: 'UNREAD' as NotificationsFilterOption,
+            lastItemId: lastPage?.notifications.at(-1)?.id,
+          }
+        : undefined;
```

Comment:
```suggestion
      if ( allpages[0]?.totalItemCount &&
        allpages[0].totalItemCount > currentCount ) {
        return {
            pageSize: NOTIFICATIONS_TAKE,
            filterOption: 'UNREAD' as NotificationsFilterOption,
            lastItemId: lastPage?.notifications.at(-1)?.id,
          }
      }
```
return 문이 조금 복잡해보여서 요렇게 간결하게 하면 어떨까요오👀

Reply:
훨씬 간결해지네 좋아요오 👍